server/src/test/java/io/crate/protocols/postgres/PostgresWireProtocolTest.java

```diff
@@ -817,8 +821,8 @@ private void submitQueriesThroughSimpleQueryMode(String statements,
                                                      @Nullable Throwable failure,
                                                      @Nullable CompletableFuture future) {
         Sessions sqlOperations = Mockito.mock(Sessions.class);
-        Session session = mock(Session.class);
-        when(session.execute(any(String.class), any(int.class), any(RowCountReceiver.class))).thenReturn(future);
+        Session session = spy(this.sqlOperations.newSystemSession());
```

Comment:
Changes in this Itest fixes the failures. The spy instance allows the actual method(handleSimpleQuery) to be called.